src/utils/auth.js

```diff
@@ -0,0 +1,51 @@
+import { processServerResponse, baseUrl } from "./utils.js";
+
+export const register = (email, password, name, avatar) => {
+  return fetch(`${baseUrl}/signup`, {
+    method: "POST",
+    headers: {
+      Accept: "application/json",
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify({ email, password, name, avatar }),
+  })
+    .then(processServerResponse)
+    .then((data) => {
+      return data;
+    })
+    .catch((err) => console.log(err));
```

Comment:
It’s better not to use `catch` in `Api` because you need to `catch` possible errors at the very **end** of any request and this end is in  (`App.js`)  after all blocks `then`. And in real projects blocks `catch` usually open error popups and so on.